### PR TITLE
Add a --linker-flags argument

### LINKED ISSUE #214: External dependencies in Jou programs
This may be the last thing I need to do before I can start working on a self-hosted compiler (#116).

Let's say you want to write a C or Jou program that uses LLVM:
- On Windows, you give a path to `LLVM-C.lib` as a command-line argument to the linker, and you make sure that a somewhat large collection of DLL files is available when the program runs. You need to download these files by either running LLVM's installer or by extracting them from LLVM's installer without running the installer.
- On Linux, before you compile the program, you need to run `sudo apt install llvm-11-dev` on terminal to install LLVM. Then you can run a terminal command `llvm-config-11 --ldflags --libs` (where `11` is the LLVM version), which prints something like `-L/usr/lib/llvm-11/lib -lLLVM-11`. These are flags that you need to give to the linker.

In other words: it is a mess, and what exactly you need to do depends on your OS. Maybe some day there will be a Jou package manager that handles the mess for you.

But for now, I think it would be good enough if you could specify linker flags in Jou code. You still need to download/install the external library some os-specific way, but at least it would be possible to use it. Maybe a `linker_flags` keyword? Something like this:

```python3
if WINDOWS:  # Code inside this would be compiled only on windows.
    linker_flags "lib/LLVM-C.lib"
else:
    linker_flags "-L/usr/lib/llvm-11/lib -lLLVM-11"

declare LLVMSomeFunction(...)

def main() -> int:
    LLVMSomeFunction(...)
    return 0
```

### PR DESCRIPTION
**Pull request by Akuli**
A bit annoying to use, but fixes #214 for now. Hopefully I will eventually create an actual package manager.

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -75,6 +76,19 @@ static void parse_arguments(int argc, char **argv, CommandLineFlags *flags, cons
         } else if (!strcmp(argv[i], "--verbose")) {
             flags->verbose = true;
             i++;
+        } else if (!strcmp(argv[i], "--linker-flags")) {
+            if (flags->linker_flags) {
+                // TODO: test
```
Comment (Akuli) [inline src/main.c:81]:
```suggestion
```

File: src/main.c [hunk 2]
```diff
@@ -75,6 +76,19 @@ static void parse_arguments(int argc, char **argv, CommandLineFlags *flags, cons
...
+                fprintf(stderr, "%s: --linker-flags cannot be given multiple times", argv[0]);
+                goto wrong_usage;
+            }
+            if (argc-i < 2) {
+                // TODO: test
```
Comment (Akuli) [inline src/main.c:86]:
```suggestion
```

File: examples/x11_window.jou [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    gc = XCreateGC(display, window, 0, NULL)
+    XSetForeground(display, gc, 0xffffffL)
+
+    XSelectInput(display, window, 32768)
+    XMapRaised(display, window)
+
+    while True:
+        usleep(100)
+        x = XDrawImageString(display, window, gc, 50, 50, "hello", 5)
```
Comment (taahol) [inline examples/x11_window.jou:57]: Variable `x` not needed.

Comment (Akuli) [reply examples/x11_window.jou:57]: Related: #26

File: tests/should_succeed/compiler_cli.jou [hunk 4]
```diff
@@ -28,6 +28,8 @@ def main() -> int:
     run_jou("-O8 examples/hello.jou")  # Output: <jouexe>: unknown argument "-O8" (try "<jouexe> --help")
     run_jou("--lolwat")  # Output: <jouexe>: unknown argument "--lolwat" (try "<jouexe> --help")
     run_jou("lolwat.jou")  # Output: compiler error in file "lolwat.jou": cannot open file: No such file or directory
+    run_jou("--linker-flags")  # Output: <jouexe>: there must be a string of flags after --linker-flags (try "<jouexe> --help")
+    run_jou("--linker-flags x --linker-flags y")  # Output: <jouexe>: --linker-flags cannot be given multiple times (try "./jou --help")
```
Comment (littlewhitecloud) [inline tests/should_succeed/compiler_cli.jou:32]: # Output: <jouexe>: --linker-flags cannot be given multiple times (try "<jouexe> --help")

File: tests/should_succeed/compiler_cli.jou [hunk 5]
```diff
@@ -27,16 +28,19 @@ def main() -> int:
...
```
Comment (littlewhitecloud) [inline tests/should_succeed/compiler_cli.jou:32]: run_jou("--linker-flags x --linker-flags y")  # Output: <jouexe>: --linker-flags cannot be given multiple times (try "./jou --help")

Comment (littlewhitecloud) [inline tests/should_succeed/compiler_cli.jou:32]: Should be:
```python
run_jou("--linker-flags x --linker-flags y")  # Output: <jouexe>: --linker-flags cannot be given multiple times (try "<jouexe> --help")
```

**Review by littlewhitecloud — CHANGES_REQUESTED** [review]

File: tests/should_succeed/compiler_cli.jou (hunk 5 above)
Comment (littlewhitecloud) [inline tests/should_succeed/compiler_cli.jou:32]: Should be:
```python
run_jou("--linker-flags x --linker-flags y")  # Output: <jouexe>: --linker-flags cannot be given multiple times (try "<jouexe> --help")
```

Comment (littlewhitecloud) [inline tests/should_succeed/compiler_cli.jou:32]:
```suggestion
    run_jou("--linker-flags x --linker-flags y")  # Output: <jouexe>: --linker-flags cannot be given multiple times (try "<jouexe> --help")
```

**Review by littlewhitecloud — CHANGES_REQUESTED** [review]

File: tests/should_succeed/compiler_cli.jou (hunk 5 above)
Comment (Akuli) [reply tests/should_succeed/compiler_cli.jou:32]: Thanks :)

**Pull request merged**